src/UKHO.S100PermitService.Common/Clients/ProductKeyServiceApiClient.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Text;
+
+namespace UKHO.S100PermitService.Common.Clients
+{
+    public class ProductKeyServiceApiClient : IProductKeyServiceApiClient
+    {
+        private readonly HttpClient _httpClient;
+
+        public ProductKeyServiceApiClient(IHttpClientFactory httpClientFactory)
+        {
+            _httpClient = httpClientFactory.CreateClient();
+        }
+
+        public async Task<HttpResponseMessage> CallProductKeyServiceApiAsync(string uri, HttpMethod httpMethod, string payLoad, string accessToken, string correlationId)
+        {
+            using var httpRequestMessage = new HttpRequestMessage(httpMethod, uri);
+            httpRequestMessage.Content = new StringContent(payLoad, Encoding.UTF8, "application/json");
+
+            if(!string.IsNullOrEmpty(accessToken))
+            {
+                httpRequestMessage.SetBearerToken(accessToken);
+                httpRequestMessage.AddHeader(PermitServiceConstants.XCorrelationIdHeaderKey, correlationId);
+            }
+
+            return await _httpClient.SendAsync(httpRequestMessage, CancellationToken.None);
```

Comment:
You should specify a cancellation token here. 

The method should specify a cancellation token as a parameter, which should be passed here. The calling methods should all pass a cancellation token up to the root, where you should have a CancellationToken in the controller signature - this is bound to the HttpContext.RequestAborted property at runtime. Could you check for other instances and amend please?

Reply:
Resolved